demo/oauth2-yelp.R

```diff
@@ -0,0 +1,31 @@
+library(httr)
+
+# This example demonstrates the use of client credentials grant
+
+# 1. Find OAuth settings for yelp:
+#    https://www.yelp.ca/developers/documentation/v3/authentication
+#    Set authorize url to NULL as we are not using Authorization code grant but
+#    client credential grant
+yelp_endpoint <- oauth_endpoint(
+  authorize = NULL,
+  access    = "https://api.yelp.com/oauth2/token"
+)
+
+# 2. Register an application at https://www.yelp.com/developers/v3/manage_app
+#    Replace key and secret below.
+yelp_app <- oauth_app("yelp", "bvmjj2EOBvOknQ", "n8ueSvTNdlE0BDDJpLljvmgUGUw")
+
+# 3. Get OAuth credentials using client credential grant
+#    Yelp do not use basic auth. Use `use_basic_auth = T` otherwise
+yelp_token <- oauth2.0_token(yelp_endpoint, yelp_app,
+                        client_credentials = T)
+
+# 4. Use API
+url <- modify_url("https://api.yelp.com",
```

Comment:
Can you please use the long-lines style from here: http://style.tidyverse.org/syntax.html#long-lines